runtime/sys.c

```diff
@@ -363,18 +363,29 @@ CAMLprim value caml_sys_chdir(value dirname)
   CAMLreturn(Val_unit);
 }
 
+static char_os* canonicalize_path(char_os *path)
+{
+#ifdef _WIN32
+  return caml_win32_dos_to_nt_path(path);
```

Comment:
For backwards compatibility could we keep any paths that are under `MAX_PATH - 12` unmodified? Pretending that `char_os` (what is that?) is compatible with `strlen`, then something like:

```c
if (strlen(path) <= MAX_PATH - 12) { return caml_stat_strdup(path); }
else { return caml_win32_dos_to_nt_path(path); }
```

note: The magic "12" constant comes from [first MSDN article](https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation): *the directory name cannot exceed MAX_PATH minus 12*.

Reply:
> The native C character type under Windows is `WCHAR`, two bytes wide, while under Unix it is `char`, one byte wide. A type `char_os` is defined in `<caml/misc.h>` that stands for the concrete C character type of each platform. Strings in the platform encoding are of type `char_os *`.

[13 Advanced topic: interfacing with Windows Unicode APIs](https://ocaml.org/manual/5.2/intfc.html#s:interfacing-windows-unicode-apis)

It is intended for platform-independent code, as are the `*_os` macros from `misc.h`.